src/huggingface_hub/hf_api.py

```diff
@@ -1148,6 +1148,100 @@ def list_metrics(self) -> List[MetricInfo]:
         d = r.json()
         return [MetricInfo(**x) for x in d]
 
+    @_deprecate_positional_args
+    def list_spaces(
+        self,
+        *,
+        filter: Union[str, Iterable[str], None] = None,
+        author: Optional[str] = None,
+        search: Optional[str] = None,
+        sort: Union[Literal["lastModified"], str, None] = None,
+        direction: Optional[Literal[-1]] = None,
+        limit: Optional[int] = None,
+        cardData: Optional[bool] = None,
+        datasets: Union[bool, str, Iterable[str], None] = None,
+        models: Union[bool, str, Iterable[str], None] = None,
+        linked: Optional[bool] = None,
+        full: Optional[bool] = None,
+        use_auth_token: Optional[str] = None,
+    ) -> List[SpaceInfo]:
+        """
+        Get the public list of all Spaces on huggingface.co
+
+        Args:
+            filter `str` or `Iterable`, *optional*):
+                A string or which can be used to identify Spaces on the hub.
```

Comment:
```suggestion
                A string tag or list of tags that can be used to identify Spaces on the Hub.
```

Maybe I would also mention it corresponds to the `tags` and `sdk` field in https://huggingface.co/spaces/evaluate-metric/precision/blob/main/README.md#L10